### PR TITLE
readme + release cleanup

### PR DESCRIPTION
**Pull request by metachris**
## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -54,11 +54,13 @@ git tag -s v2.3.1-rc1  # without a tag, the Docker image would include the wrong
 # now push to Github (CI will build the Docker image: https://github.com/flashbots/mev-boost/actions)
 git push origin --tags
 
-# you can also manually create and push the Docker image
+# all done here
+#
+# if you want manually create and push the Docker image:
```
Comment (ralexstokes) [inline RELEASE.md:59]:
```suggestion
# if you want to manually create and push the Docker image:
```

File: RELEASE.md [hunk 2]
```diff
@@ -54,11 +54,13 @@ git tag -s v2.3.1-rc1  # without a tag, the Docker image would include the wrong
 # now push to Github (CI will build the Docker image: https://github.com/flashbots/mev-boost/actions)
 git push origin --tags
 
-# you can also manually create and push the Docker image
+# all done here
```
Comment (ralexstokes) [inline RELEASE.md:57]: "all done here" ?

Comment (metachris) [reply RELEASE.md:57]: yep, all is done here :)

**Review by jtraglia — APPROVED** [review]
LGTM

**Pull request merged**